ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
Cool (learned something new SGTM).

@jacobbednarz  - Any chance you can update the patch to not touch private.  Also, any reason for the macros versus just a function call?